src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -37,14 +40,48 @@ public ConfigurationReader(IConfigurationSection configuration, DependencyContex
 
         public void Configure(LoggerConfiguration loggerConfiguration)
         {
-            ApplyMinimumLevel(loggerConfiguration);
-            ApplyEnrichment(loggerConfiguration);
-            ApplyFilters(loggerConfiguration);
-            ApplySinks(loggerConfiguration);
-            ApplyAuditSinks(loggerConfiguration);
-        }        
-
-        void ApplyMinimumLevel(LoggerConfiguration loggerConfiguration)
+            var declaredLevelSwitches = ProcessLevelSwitchDeclarations();
+            ApplyMinimumLevel(loggerConfiguration, declaredLevelSwitches);
+            ApplyEnrichment(loggerConfiguration, declaredLevelSwitches);
+            ApplyFilters(loggerConfiguration, declaredLevelSwitches);
+            ApplySinks(loggerConfiguration, declaredLevelSwitches);
+            ApplyAuditSinks(loggerConfiguration, declaredLevelSwitches);
+        }
+
+        private IReadOnlyDictionary<string, LoggingLevelSwitch> ProcessLevelSwitchDeclarations()
```

Comment:
Very minor nitpick- the Serilog repos don't use explicit `private` or `internal` (I think there might be a DotSettings file in the main repo.)

Reply:
yes indeed :) I'll copy over the `.DotSettings` file from the main repo to detect those cases.